saleor/graphql/webhook/subscription_types.py

```diff
@@ -38,7 +44,80 @@
 }
 
 
-class AppBase(AbstractType):
+class IssuingPrincipalType(Enum):
+    APP = "app"
+    USER = "user"
+
+
+class IssuingPrincipal(ObjectType):
```

Comment:
IssuingPrincipal could be implemented as union, see usages of `graphene.Union` and [the docs here](https://graphql.org/learn/schema/#union-types).